apps/src/dance/Dance.js

```diff
@@ -196,6 +230,11 @@ Dance.prototype.onReportComplete = function (response) {
  * Click the run button.  Start the program.
  */
 Dance.prototype.runButtonClick = function () {
+  //If the audio is still loading, no action
+  if (!Sounds.getSingleton().isLoaded(getStore().getState().selectedSong)) {
+    return;
+  }
+
```

Comment:
This seems odd. I thought we could just start when the user pressed Run and then wait until the song was fully loaded before animating. That's how the prototype I created worked with the playSound() callback.

Reply:
This was my bad, I suggested following the pattern we use in Craft (that just bails if the runtime isn't loaded yet).

https://github.com/code-dot-org/code-dot-org/blob/67eb8face9fcea5e287edd91c21a85170b38f761/apps/src/craft/simple/craft.js#L561-L564

If we delay the start, would we need to add a spinner and protect against multiple clicks?  Prefer to address this in a follow-up PR.

Reply:
Ah, I didn't read your top-level review comment first.  What you're saying makes sense if we start p5 right away, but only draw a loader until the callback is fired!

We should go with whatever is simpler to get this set of changes merged for now...

Reply:
Since this song could take a while to load over slow connections, it seems like we should have "dead" Run button presses where nothing happens. I'm happy to help as needed to get this in - just let me know...

Reply:
Also, I don't think a loader is really required - the Run button will at least toggle to Reset and the screen can still still have a background effect active since running will have started. It's just that the dancer moves shouldn't start until the song is playing...